test/library/standard/Random/RandomStreamInterface/choiceTestUtils.chpl

```diff
@@ -6,7 +6,11 @@ proc test(stream, X, size:?sizeType=none, replace=true, prob:?probType=none, tri
   var counts;
   if isArrayValue(X) {
     counts = new map(X.eltType, int);
-  } else {
+  }
+  else if X.rank > 1 {
```

Comment:
Could you at least add a comment here saying that if we're hitting this `if`, `X` must be domain? You can also make the code more self-documenting by 

```chapel
else if isDomainValue(x) {
  if X.rank > 1 {

  }
  else {

  }
}
else {
  compilerError("Unexpected type");
}